internal/cli/atlas/serverlessclusters/create.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serverlessclusters
+
+import (
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const providerName = "SERVERLESS"
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	clusterName string
+	provider    string
+	region      string
+	store       store.ServerlessInstanceCreator
+}
+
+func (opts *CreateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+var createTemplate = "Serverless Cluster {{.Name}} created.\n"
+
+func (opts *CreateOpts) Run() error {
+	r, err := opts.store.CreateServerlessInstance(opts.ConfigProjectID(), opts.newServerlessCreateRequestParams())
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *CreateOpts) newServerlessCreateRequestParams() *atlas.ServerlessCreateRequestParams {
+	return &atlas.ServerlessCreateRequestParams{
+		Name: opts.clusterName,
+		ProviderSettings: &atlas.ServerlessProviderSettings{
+			BackingProviderName: opts.provider,
+			ProviderName:        providerName,
+			RegionName:          opts.region,
+		},
+	}
+}
+
+// mongocli atlas serverlessCluster(s)|sc create <clusterName> --backingProviderName backingProviderName --providerName providerName --regionName regionName [--projectId projectId].
+func CreateBuilder() *cobra.Command {
```

Comment:
[late to the party q] How do we ensure these commands only exist on Atlas and don't appear on the OM/CM cli?

Reply:
Afaik, there are separate directories in `internal/cli` directory. `atlas, opasmanager, cloudmanager`. Each of them has a set of sub commands which are available under them:

Example for `cloudmanager`:
```
	cmd.AddCommand(
		clusters.Builder(),
		alerts.Builder(),
		backup.Builder(),
		servers.Builder(),
		automation.Builder(),
		security.Builder(),
		dbusers.Builder(),
		events.Builder(),
		monitoring.Builder(),
		processes.Builder(),
		metrics.Builder(),
		logs.Builder(),
		agents.Builder(),
		maintenance.Builder(),
		performanceadvisor.Builder(),
		featurepolicies.Builder(),
```